common/lib/capa/capa/tests/test_capa_problem.py

```diff
@@ -0,0 +1,163 @@
+"""
+Test capa problem.
+"""
+import unittest
+
+from . import new_loncapa_problem
+
+
+class CAPAProblemTest(unittest.TestCase):
+    """ CAPA problem related tests"""
+
+    def test_label_and_description_inside_responsetype(self):
+        """
+        Verify that
+        * label is extracted
+        * description_ids are constructed
+        * <label> tag is removed to avoid duplication
+
+        This is the case when we have a problem with single question or
+        problem with multiple-questions separated as per the new format.
+        """
+        xml = """
+        <problem>
+            <choiceresponse>
+                <label>Select the correct synonym of paranoid?</label>
+                <description>Only the paranoid survive.</description>
+                <checkboxgroup>
+                    <choice correct="true">over-suspicious</choice>
+                    <choice correct="false">funny</choice>
+                </checkboxgroup>
+            </choiceresponse>
+        </problem>
+        """
+        problem = new_loncapa_problem(xml)
+        self.assertEqual(
+            problem.problem_data,
+            {
+                '1_2':
+                {
+                    'description_ids': '1_description_2_1',
+                    'label': 'Select the correct synonym of paranoid?',
+                    'descriptions': {'1_description_2_1': 'Only the paranoid survive.'}
+                }
+            }
+        )
+        self.assertEqual(len(problem.tree.xpath('//label')), 0)
+
+    def test_label_attribute_only(self):
+        """
+        Verify that label is extracted and <p> tag with question
+        text is removed when label attribute is set on inputtype.
+
+        This is the case when we have a OLX only problem with multiple-questions.
+        """
+        question = "Once we become predictable, we become ______?"
+        xml = """
+        <problem>
+            <p>Be sure to check your spelling.</p>
+            <p>{}</p>
+            <stringresponse answer="vulnerable" type="ci">
+                <textline label="{}" size="40"/>
+            </stringresponse>
+        </problem>
+        """.format(question, question)
+        problem = new_loncapa_problem(xml)
+        self.assertEqual(
+            problem.problem_data,
+            {'1_2': {'description_ids': '', 'label': question, 'descriptions': {}}}
+
+        )
+        self.assertEqual(
+            len(problem.tree.xpath('//p[text()="{}"]'.format(question))),
+            0
+        )
+
+    def test_neither_label_tag_nor_attribute(self):
+        """
+        Verify that label is extracted correctly.
+
+        This is the case when we have a markdown problem with multiple-questions.
+        In this case when markdown is converted to xml, there will be no label
+        tag and label attribute inside responsetype. But we have a label tag
+        before the responsetype.
```

Comment:
label attribute is always set on inputtype and inputtype always come inside responsetype so we can't have label attribute outside responsetype.